src/TreePiece.h

```diff
@@ -566,4 +569,47 @@ void TreePiece<Data>::print(Node<Data>* node) {
   out.close();
 }
 
+template <typename Data>
+void TreePiece<Data>::output(std::string output_file, const CkCallback& cb) {
+  FILE* fp;
+  // Print total number of particles
+  if (this->thisIndex == 0 && dim_cnt == 0) {
+    fp = CmiFopen(output_file.c_str(), "w");
+    CkAssert(fp);
+    fprintf(fp, "%d\n", n_total_particles);
+    CmiFclose(fp);
+  }
+
+  // Print particle accelerations to output file
+  fp = CmiFopen(output_file.c_str(), "a");
+  CkAssert(fp);
+  for (auto leaf : leaves) {
+    for (int i = 0; i < leaf->n_particles; i++) {
+      Particle& particle = leaf->particles[i];
+      Real outval;
+      if (dim_cnt == 0) outval = particle.acceleration.x;
+      else if (dim_cnt == 1) outval = particle.acceleration.y;
+      else if (dim_cnt == 2) outval = particle.acceleration.z;
+      fprintf(fp, "%.14g\n", outval);
+    }
+  }
+
+  if (++dim_cnt == 3) dim_cnt = 0;
```

Comment:
Might consider writing this as `dim_cnt = (dim_cnt + 1) % 3`

Reply:
Good point